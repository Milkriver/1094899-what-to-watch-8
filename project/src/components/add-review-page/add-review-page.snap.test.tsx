import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';
import AddReviewPage from './add-review-page';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';


describe('Component: AddReviewPage', () => {
  const mockStore = configureMockStore();

  const store = mockStore({
    MOVIES: {
      movie: {
        id: 1, name: '', posterImage: '', previewImage: '', backgroundImage: '',
        backgroundColor: '', videoLink: '', previewVideoLink: '', description: '', rating: 1, scoresCount: 1, director: '', starring: ['', ''],
        runTime: 1, genre: '', released: 1, isFavorite: true,
      },
    },
    USER: {},
    REVIEWS: {},
  });

  store.dispatch = jest.fn();

  test('should render correctly', () => {
    const history = createMemoryHistory();
    const page = (
      <Provider store={store}>
        <Router history={history}>
          <AddReviewPage />;
        </Router>
      </Provider>
    );
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
