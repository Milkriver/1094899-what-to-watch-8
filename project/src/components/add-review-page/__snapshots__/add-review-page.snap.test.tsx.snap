// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddReviewPage should render correctly 1`] = `
<div>
  <section
    class="film-card film-card--full"
  >
    <div
      class="film-card__header"
    >
      <div
        class="film-card__bg"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          class="breadcrumbs"
        >
          <ul
            class="breadcrumbs__list"
          >
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="/films//"
              />
            </li>
            <li
              class="breadcrumbs__item"
            >
              <div
                class="breadcrumbs__link"
              >
                 Add review
              </div>
            </li>
          </ul>
        </nav>
        <div
          class="user-block"
        >
          <a
            class="user-block__link"
            href="/login"
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        class="film-card__poster film-card__poster--small"
      >
        <img
          alt=""
          height="327"
          src=""
          width="218"
        />
      </div>
    </div>
    <div
      class="add-review"
    >
      <form
        action="#"
        class="add-review__form"
      >
        <div
          class="rating"
        >
          <div
            class="rating__stars"
          >
            <input
              class="rating__input"
              id="star-10"
              name="rating"
              required=""
              type="radio"
              value="10"
            />
            <label
              class="rating__label"
              for="star-10"
            >
               Rating 
              10
            </label>
            <input
              class="rating__input"
              id="star-9"
              name="rating"
              required=""
              type="radio"
              value="9"
            />
            <label
              class="rating__label"
              for="star-9"
            >
               Rating 
              9
            </label>
            <input
              checked=""
              class="rating__input"
              id="star-8"
              name="rating"
              required=""
              type="radio"
              value="8"
            />
            <label
              class="rating__label"
              for="star-8"
            >
               Rating 
              8
            </label>
            <input
              class="rating__input"
              id="star-7"
              name="rating"
              required=""
              type="radio"
              value="7"
            />
            <label
              class="rating__label"
              for="star-7"
            >
               Rating 
              7
            </label>
            <input
              class="rating__input"
              id="star-6"
              name="rating"
              required=""
              type="radio"
              value="6"
            />
            <label
              class="rating__label"
              for="star-6"
            >
               Rating 
              6
            </label>
            <input
              class="rating__input"
              id="star-5"
              name="rating"
              required=""
              type="radio"
              value="5"
            />
            <label
              class="rating__label"
              for="star-5"
            >
               Rating 
              5
            </label>
            <input
              class="rating__input"
              id="star-4"
              name="rating"
              required=""
              type="radio"
              value="4"
            />
            <label
              class="rating__label"
              for="star-4"
            >
               Rating 
              4
            </label>
            <input
              class="rating__input"
              id="star-3"
              name="rating"
              required=""
              type="radio"
              value="3"
            />
            <label
              class="rating__label"
              for="star-3"
            >
               Rating 
              3
            </label>
            <input
              class="rating__input"
              id="star-2"
              name="rating"
              required=""
              type="radio"
              value="2"
            />
            <label
              class="rating__label"
              for="star-2"
            >
               Rating 
              2
            </label>
            <input
              class="rating__input"
              id="star-1"
              name="rating"
              required=""
              type="radio"
              value="1"
            />
            <label
              class="rating__label"
              for="star-1"
            >
               Rating 
              1
            </label>
          </div>
        </div>
        <div
          class="add-review__text"
        >
          <textarea
            class="add-review__textarea"
            id="review-text"
            maxlength="400"
            name="review-text"
            placeholder="Review text"
            required=""
          />
          <div
            class="add-review__submit"
          >
            <button
              class="add-review__btn"
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
  ;
</div>
`;
